test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightBatchReaderTests.cs

```diff
@@ -363,6 +363,40 @@ public async Task ReadBatchRequestWithChangesetAndDependsOnIdsAsync()
                 isResponse: false);
         }
 
+        [Fact]
+        public async Task ReadBatchRequestWithNullHeadersAsync()
+        {
+            var payload = "{\"requests\": [{" +
+                "\"id\": \"1\"," +
+                "\"method\": \"POST\"," +
+                "\"url\": \"http://tempuri.org/Customers\"," +
+                "\"headers\": {\"odata-version\":\"4.0\",\"content-type\":\"application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8\",\"null-header\":null}, " +
+                "\"body\": {\"@odata.type\":\"#NS.Customer\",\"Id\":1,\"Name\":\"Customer 1\",\"Type\":\"Retail\"}}]}";
+
+            await SetupJsonLightBatchReaderAndRunTestAsync(
+                payload,
+                async (jsonLightBatchReader) =>
+                {
+                    try
+                    {
+                        while (await jsonLightBatchReader.ReadAsync())
+                        {
+                            if (jsonLightBatchReader.State == ODataBatchReaderState.Operation)
+                            {
+                                await jsonLightBatchReader.CreateOperationRequestMessageAsync();

```

Comment:
What is the expected value of the parsed `null-header`? I think you should have assertions that verify your intentions, i.e. if the headers you expect in the request message actually exist in the request message with the expected values.

Reply:
@habbes, I've specified the purpose of this test case and improved readability.
That test case should verify that an json property with NULL value won't cause an HTTP 500 internal server error.
I hope the improved procedure is easier to understand.
Secondly I've added additional test cases to verify, that duplicate json properties return in a correct odata error, and not HTTP 500 internal server error.

Would you please check them again?